packages/auth-construct/src/construct.ts

```diff
@@ -900,132 +902,209 @@ export class AmplifyAuth
+        (cfnUserPool.enabledMfas ?? []).forEach((type) => {
+          if (type === 'SMS_MFA') {
+            mfaTypes.push('SMS');
+          }
+          if (type === 'SOFTWARE_TOKEN_MFA') {
+            mfaTypes.push('TOTP');
+          }
+        });
+        return JSON.stringify(mfaTypes);
+      },
+    });
 
-    if (this.computedUserPoolProps.mfa) {
-      output.mfaConfiguration = this.computedUserPoolProps.mfa;
+    // extract social providers from UserPool resource
+    output.socialProviders = Lazy.string({
+      produce: () => {
+        const outputProviders = [];
+        const userPoolProviders = this.resources.userPool.identityProviders;
+        if (userPoolProviders) {
+          for (const provider of userPoolProviders) {
+            const providerType =
+              provider.node['_children']['Resource']['providerType'];
 
-      const mfaTypes = [];
-      if (this.computedUserPoolProps.mfaSecondFactor?.otp) {
-        mfaTypes.push('TOTP');
-      }
-      if (this.computedUserPoolProps.mfaSecondFactor?.sms) {
-        mfaTypes.push('SMS');
-      }
-      if (mfaTypes.length > 0) {
-        output.mfaTypes = JSON.stringify(mfaTypes);
-      }
-    }
+            if (providerType === 'Google') {
+              outputProviders.push('GOOGLE');
+            }
+            if (providerType === 'Facebook') {
+              outputProviders.push('FACEBOOK');
+            }
+            if (providerType === 'SignInWithApple') {
+              outputProviders.push('SIGN_IN_WITH_APPLE');
+            }
+            if (providerType === 'LoginWithAmazon') {
+              outputProviders.push('LOGIN_WITH_AMAZON');
+            }
+            if (providerType === 'OIDC') {
+              outputProviders.push(
+                provider.node['_children']['Resource']['providerName']
+              );
+            }
+            if (providerType === 'SAML') {
+              outputProviders.push(
+                provider.node['_children']['Resource']['providerName']
+              );
+            }
+          }
+          if (outputProviders.length > 0) {
+            return JSON.stringify(outputProviders);
+          }
+        }
+        return undefined;
+      },
+    });
 
-    if (this.providerSetupResult.providersList.length > 0) {
-      output.socialProviders = JSON.stringify(
-        this.providerSetupResult.providersList
-      );
-    }
-    // if callback URLs are configured, we must expose the oauth settings to the output
-    if (
-      this.providerSetupResult.oAuthSettings &&
-      this.providerSetupResult.oAuthSettings.callbackUrls
-    ) {
-      const oAuthSettings = this.providerSetupResult.oAuthSettings;
-      if (this.domainPrefix) {
-        output.oauthCognitoDomain = `${this.domainPrefix}.auth.${
-          Stack.of(this).region
-        }.amazoncognito.com`;
-      }
+    const getUserPoolClientResource = () => {
+      return this.resources.userPoolClient.node['_children']['Resource'];
```

Comment:
We shouldn't do that. An `_` typically represents an internal variable/API that is not meant for consumption by clients, i.e. cdk might change this or behavior without requiring a Major version.

Reply:
good point; I will try to find another way to extract providerType

Reply:
I found a way to do this with findResource and casting to the resource type